src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -98,6 +109,33 @@ internal static class ParameterHelpers
             return parameters;
         }
 
+        private static bool IsPropertyWithoutBody(Symbol owner)
+        {
+            if (owner is SourcePropertySymbol)
+            {
+                var property = (SourcePropertySymbol)owner;
+                var getterWithoutBody = property.GetMethod == null || IsAnyMethodWithoutBody(property.GetMethod);
+                var setterWithoutBody = property.SetMethod == null || IsAnyMethodWithoutBody(property.SetMethod);
+                return getterWithoutBody && setterWithoutBody;
+            }
+
+            return false;
+        }
+
+        private static bool IsAnyMethodWithoutBody(Symbol owner)
+        {
+            if (owner is SourceMethodSymbol)
+            {
+                var method = (SourceMethodSymbol)owner;
+                if (method.BodySyntax == null)
```

Comment:
Yes.  The presence/absence of this is how we determine if we should report an error about 'async' on methods without bodies.  That's the code i was basing this check off.  Since async/readonly are similar in that they only affect the body of a method, it seemed the right way to check.